src/broadcast/broadcast.rs

```diff
@@ -291,7 +292,8 @@ impl<N: NodeIdT> Broadcast<N> {
 
     fn handle_echo_hash(&mut self, sender_id: &N, hash: &Digest) -> Result<Step<N>> {
         // If the sender has already sent `EchoHash`, ignore.
-        if let Some(old_hash) = self.echo_hashes.get(sender_id) {
+        // Should there be an error case for sending an `EchoHash` after an `Echo` (2nd element not None)
```

Comment:
It's debatable whether we should expect messages from a single peer to arrive in the right order. The algorithms should work fine if we don't. 
For now, I wouldn't consider this a fault, as long as the hashes match.
However, we need to make sure we don't insert `None` again below, if we already have a `Some(_)`!